pypiper/utils.py

```diff
@@ -292,14 +287,25 @@ def flag_name(status):
     """
     Determine the name for a flag file of the status indicated.
 
-    :param status: Name of status for which to create flag file name.
-    :type status: str
-    :return: Name of flag file corresponding to given status.
-    :rtype: str
+    :param str status: Name of status for which to create flag file name.
+    :return str: Name of flag file corresponding to given status.
     """
     return status + ".flag"
 
 
+def get_proc_name(cmd):
+    """
+    Get the representative process name from complex command
+
+    :param str | list[str] cmd: a command to be processed
+    :return str: the basename representative command
+    """
+    if isinstance(cmd, list):
```

Comment:
Can we make this `Iterable` so that we accept tuples also?

Reply:
Fine with me -- will we be giving it tuples? I'm not do you just change `list` to `Iterable` ?